usr/libexec/playtron/hardware-test-tool

```diff
@@ -638,7 +638,53 @@ class StressTest(PlaceholderTest):
                 self.stress_time = -1
 
 
+BAUD_RATE = 115200
+DEVICE = "/dev/ttyS2"
+DATA = b'\xE7\x00\x00\x22\x02\x00\x00\x00\x0F\x33\xED'
+EXPECTED_RESPONSE = b'\xE7\x55\xAA\x01\xED'
+class SerialTest(PlaceholderTest):
+    def __init__(self):
+        self.name = 'Serial Test|串行端口'
+        self.reset()
+        self.start_time = None
+
+    def start(self, screen, font):
+        self.reset()
+        self.start_time = time.perf_counter()
+
+        # Configure the serial port
+        cmd = "stty -F {device} {baud_rate}".format(device=DEVICE, baud_rate=BAUD_RATE)
+        if os.system(cmd) != 0:
+            self.result = "fail"
+            return
+        cmd = "stty -F {device} raw".format(device=DEVICE)
+        if os.system(cmd) != 0:
+            self.result = "fail"
+            return
+
+        # Open the serial interface
+        reader = open(DEVICE, "rb")
+        writer = open(DEVICE, "wb")
+
+        # Write a payload
+        writer.write(DATA)
+        writer.close()
 
+        # Read the response
+        response = reader.read(5)
+        if response == EXPECTED_RESPONSE:
+            self.result = "pass"
+        else:
+            self.result = "fail"
+        reader.close()
+
+    def draw(self, screen, font):
```

Comment:
Is this even necessary? Wouldn't this test immediately go to the common pass/fail screen?

Reply:
No, if I remove the `draw` method, it goes to a blank screen.

Reply:
I can set `done = True` though, if we want.

Reply:
Oh, right. I conflated `self.result` with `self.done`. Nevermind!

I think either way works, I suppose this way the user has time to confirm the result.